src/test/scala/uk/gov/nationalarchives/consignmentexport/MainSpec.scala

```diff
@@ -0,0 +1,58 @@
+package uk.gov.nationalarchives.consignmentexport
+
+import java.io.File
+import java.util.UUID
+
+import com.github.tomakehurst.wiremock.stubbing.ServeEvent
+import uk.gov.nationalarchives.consignmentexport.Utils.PathUtils
+
+import scala.io.Source
+import scala.sys.process._
+import scala.jdk.CollectionConverters._
+
+class MainSpec extends ExternalServiceSpec {
+
+  "the export job" should "export the correct tar and checksum file" in {
+    val consignmentId = UUID.fromString("50df01e6-2e5e-4269-97e7-531a755b417d")
+    putFile(s"$consignmentId/7b19b272-d4d1-4d77-bf25-511dc6489d12")
+    Main.run(List("export", "--consignmentId", consignmentId.toString)).unsafeRunSync()
+    val objects = outputBucketObjects().map(_.key())
+
+    objects.size should equal(2)
+    objects.head should equal(s"$consignmentId.tar.gz")
+    objects.last should equal(s"$consignmentId.tar.gz.sha256")
+
+    wiremockGraphqlServer.getAllServeEvents.asScala.map(_.getRequest.getBodyAsString).foreach(println)
+  }
+
+  "the export job" should "export a valid tar and checksum file" in {
+    val consignmentId = UUID.fromString("50df01e6-2e5e-4269-97e7-531a755b417d")
+    putFile(s"$consignmentId/7b19b272-d4d1-4d77-bf25-511dc6489d12")
+    Main.run(List("export", "--consignmentId", consignmentId.toString)).unsafeRunSync()
+    val path = s"src/test/resources/testfiles/$consignmentId"
+    getObject(s"$consignmentId.tar.gz", s"$path/result.tar.gz".toPath)
+    getObject(s"$consignmentId.tar.gz.sha256", s"$path/result.tar.gz.sha256".toPath)
+
+    val exitCode = Seq("sh", "-c", s"tar -tf $path/result.tar.gz > /dev/null").!
+    exitCode should equal(0)
+
+    val source = Source.fromFile(new File(s"$path/result.tar.gz.sha256"))
+    val checksum = source.getLines().toList.head.split(" ").head
+
+    //We can't check the file checksum because tar uses a timestamp to gzip so it's different every time
```

Comment:
Out of curiosity, is there a different way of calculating the checksum that would give us some confidence that it's correct? e.g. is it easy to do it with the same Java API that the checksum file check uses?

Reply:
It's not the checksum calculation that's the problem, it's `tar`. When `tar` does the gzip, it uses a timestamp so the file is different each time it's created which is why the checksum is different.
I've managed to use tar piped into gzip which returns the same checksum for a tar file on the same set of files. I got the idea from here https://stackoverflow.com/questions/32997526/how-to-create-a-tar-file-that-omits-timestamps-for-its-contents